.github/workflows/release.yml

```diff
@@ -0,0 +1,98 @@
+        required: true
+        type: choice
+        options:
+          - "true"
+          - "false"
+        default: "false"
+      publishToOVSX:
+        description: "Publish to OpenVSX Registry ?"
+        required: true
+        type: choice
+        options:
+          - "true"
+          - "false"
+        default: "false"
+jobs:
+  packaging-job:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout vscode-knative
+        uses: actions/checkout@v4
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - name: Install dependencies
+        run: |
+          npm install -g typescript "@vscode/vsce" "ovsx"
+          npm install
+          echo "EXT_VERSION=$(cat package.json | jq -r .version)" >> $GITHUB_ENV
+          npm run vscode:prepublish
+      - name: Build And Run Unit Tests
+        uses: coactions/setup-xvfb@b6b4fcfb9f5a895edadc3bc76318fae0ac17c8b3 #v1.0.1
+        with:
+          run: npm run test
+      - name: Run UI Tests
+        uses: coactions/setup-xvfb@b6b4fcfb9f5a895edadc3bc76318fae0ac17c8b3 #v1.0.1
+        with:
+          continue-on-error: true
+          run: npm run base-ui-test
+      - name: Package
+        run: |
+          jq --tab '.extensionDependencies += [ "ms-kubernetes-tools.vscode-kubernetes-tools" ]' package.json  > package.json.new
+          mv package.json.new package.json
+          node ./out/build/update-readme.js
+          vsce package -o vscode-knative-${{ env.EXT_VERSION }}-${GITHUB_RUN_NUMBER}.vsix
+          sha256sum *.vsix > vscode-knative-${{ env.EXT_VERSION }}-${GITHUB_RUN_NUMBER}.vsix.sha256
+          ls -lash *.vsix *.sha256
+      - name: Upload VSIX Artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: vscode-knative
+          path: vscode-knative-${{ env.EXT_VERSION }}-${{github.run_number}}.vsix
+          if-no-files-found: error
+      - name: Publish to GH Release Tab
+        if: ${{ inputs.publishToMarketPlace == 'true' && inputs.publishToOVSX == 'true' }}
+        uses: "marvinpinto/action-automatic-releases@919008cf3f741b179569b7a6fb4d8860689ab7f0"
+        with:
+          repo_token: "${{ secrets.GITHUB_TOKEN }}"
+          automatic_release_tag: "${{ env.EXT_VERSION }}"
+          draft: true
+          files: |
+            vscode-knative-${{ env.EXT_VERSION }}-${{github.run_number}}.vsix
+            vscode-knative-${{ env.EXT_VERSION }}-${{github.run_number}}.vsix.sha256
+
+  release-job:
+    if: ${{ inputs.publishToMarketPlace == 'true' || inputs.publishToOVSX == 'true' }}
+    environment: ${{ 'release' }}
+    runs-on: ubuntu-latest
+    needs: packaging-job
+    steps:
+      - name: Checkout vscode-knative
+        uses: actions/checkout@v4
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - name: Install dependencies
+        run: |
+          npm install -g typescript "@vscode/vsce" "ovsx"
```

Comment:
@msivasubramaniaan there should be an `npm install` after this line. N oneed for the `vscode:prepublish`. I would remove those.